pallets/msa/src/lib.rs

```diff
@@ -1329,7 +1329,13 @@ impl<T: Config> Pallet<T> {
 
 		let mut schema_list = Vec::new();
 		for (schema_id, revoked_at) in schema_permissions {
-			schema_list.push(SchemaGrant { schema_id, revoked_at });
+			let mut actual_revoked_at = revoked_at;
+			if provider_info.revoked_at > T::BlockNumber::zero() &&
+				(revoked_at > provider_info.revoked_at || revoked_at == T::BlockNumber::zero())
+			{
+				actual_revoked_at = provider_info.revoked_at;
+			}
+			schema_list.push(SchemaGrant { schema_id, revoked_at: actual_revoked_at });
```

Comment:
A bit of code golf, but also trying to remove the mutation.

I suggest doing the push in the if-else. (Or set the `actual_revoked_at` from the if statement). Not sure if it is worth pulling out the conditional to a boolean, but it looks fine for me.

```suggestion
			if provider_info.revoked_at > T::BlockNumber::zero() &&
				(revoked_at > provider_info.revoked_at || revoked_at == T::BlockNumber::zero())
			{
				schema_list.push(SchemaGrant { schema_id, revoked_at: provider_info.revoked_at });
			} else {
    			    schema_list.push(SchemaGrant { schema_id, revoked_at });
			}
```